handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+1. Install `nislmigrate` on your new SystemLink Server.
+
+1. Attach the `D:\` volume used to capture data from your original SystemLink server.
+
+1. Run the command `nislmigrate restore --all --secret <your secret> --dir D:\migration`
+
+    !!!note
+        After this step SystemLink will migrate your test steps, results, and products from MongoDB to PostGreSQL. Depending on the size of your data set this process may take some time.
+
+1. Verify your new SystemLink Server has all the expected migrated data.
+
+## Upgrading Multi Node Configurations
+
+The following workflow describes upgrading a SystemLink Server instance that has been configured to use a dedicated MongoDB server or replica set, a dedicated PostGreSQL server or replica set, and a dedicated file store such as AWS S3.
+
+1. Backup your SystemLink Server, MongoDB server or replica set, PostGreSQL server or replica set, and file store.
+
+    !!!note
+        If S3 is used as your file store the backup step is not needed since this managed internally by AWS.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --tags --dir D:\migration`.
+
+    !!!note
+        While most of the data is external to SystemLink, tag current values are stored on the app server. Therefore this `nislmigrate` argument flag must be used to ensure that data is not lost during the migration.
+
+1. Detach `D:\`.
+
+    !!!note
+        At this point in the workflow your original SystemLink Server is in an operable state. Be aware if any new data is created or ingested by the SystemLink server at this time it will not be available to your new server.
+
+1. Provision a new Windows server for SystemLink.
+
+1. Provision a MongoDB server or replica set from the backup previously created.
+
+1. Provision a PostGres server or replica set from the backup previously created.
+
+    !!!note
+        The previous steps prescribe using a newly created instance of your MongoDB or PostGreSQL servers. This is because SystemLink may change the internal schema of these databases when services start post upgrade. If you do not start with a instance from a backup you will be unable to revert the MongoDB collections and PostGreSQL tables into the schemas needed for the older version of SystemLink.
+
+1. Install and configure the new version of SystemLink Server.
+
+1. Configure SystemLink to use the newly created MongoDB server or replica set, PostGreSQL server or replica set, and existing file store.
+
+1. Install `nislmigrate` on your new SystemLink Server.
+
+1. Attach the `D:\` volume used to capture data from your original SystemLink server.
+
+1. Run the command `nislmigrate restore --tags --dir D:\migration`.
+
+1. Verify your new SystemLink Server has all the expected migrated data.
```

Comment:
It's a bit weird that we have separate sections for migrating to remote Mongo, remote files, and remote Postgres, but only a single section on migrating multi-node to multi-node? Should we encourage users to move all the data storage off the server for deployment? Do we need to talk about multi-node migrations where only part of the data is remote?

Specifically, we did work in the files service to allow you to migrate the files database with files stored on S3 or some other external location. We can migrate only the metadata in MongoDB without migrating files. Users will probably figure this out since trying to migrate files with files stored in S3 will just fail.

Reply:
You're correct that we want people to store all data off the app server. I'll add a note regarding that expectation, which is why I don't have a distinct section for each individual data store.